homeassistant/components/hunterdouglas_powerview/config_flow.py

```diff
@@ -36,44 +37,69 @@ async def validate_input(hass: core.HomeAssistant, hub_address: str) -> dict[str
 
     try:
         async with asyncio.timeout(10):
-            device_info = await async_get_device_info(pv_request, hub_address)
+            hub = Hub(pv_request)
+            await hub.query_firmware()
+            if hub.role != "Primary":
+                raise PvApiConnectionError(
+                    f"{hub.name} ({hub.hub_address}) is performing role of {hub.role} Hub. Only the Primary Hub can manage shades"
+                )
```

Comment:
It would be better to make a new named except like `CannotConnect` and let it be trapped below so we can give them a better error in the UI. Right now they would get a cannot connect error

Reply:
have added ``UnsupportedDevice`` to the code